### PR TITLE
Convert maintainer file entries to attributes, add github handles

### PR DESCRIPTION
**Pull request by Profpatsch**
Based on https://github.com/NixOS/nixpkgs/pull/34842, the
nix-instantiate output was pretty-printed and the validity of the github handles
manually verified, by automatically checking whether the user handles exist on
github (https://github.com/userhandle, status 200 or 404).
Each handle under 5 characters was manually checked (because the collision
probability with non-maintainer accounts is high), each missing entry was
manually researched.

The script used is kept in `maintainers/scripts` as an example of how to work
with the mainainers list through nix’ JSON interface.

Hydra might need to be patched.

cc @FRidh @vcunat

### CODE REVIEW

**Comment by FRidh** [comment]
Thanks for the effort. Note the merge conflict though.

File: lib/maintainers.nix [hunk 1]
```diff
@@ -1,799 +1,3946 @@
...
+    name = "Dipin Hora";
+  };
+  disassembler = {
+    email = "disasm@gmail.com";
+    github = "disassembler";
+    name = "Samuel Leathers";
+  };
+  dizfer = {
+    email = "david@izquierdofernandez.com";
+    githu
```
Comment (cillianderoiste) [inline lib/maintainers.nix]: My github id is cillianderoiste, thanks!

Comment (Profpatsch) [reply lib/maintainers.nix]: Thanks, fixed.

**Comment by Profpatsch** [comment]
> Note the merge conflict though.

Yeah, I expect this to take a while to merge, so there might be multiple changes to maintainers nix by then.

**Comment by FRidh** [comment]
Right. That's why I suggested merging https://github.com/NixOS/nixpkgs/pull/34842 first, then fixing Hydra, and when that is done convert to what you have here.

File: lib/maintainers.nix (hunk 1 above)
Comment (jtojnar) [inline lib/maintainers.nix]: Should be @gavinrogers, see https://github.com/NixOS/nixpkgs/pull/35714

Comment (jtojnar) [inline lib/maintainers.nix]: @bignaux

Comment (jtojnar) [inline lib/maintainers.nix]: @edolstra

Comment (Profpatsch) [reply lib/maintainers.nix]: Something is broken with these markers, every one points to `dizfer` somehow?

**Comment by Profpatsch** [comment]
It looks like cross-checking entries <5 chars is not enough, normal dictionary words will most likely also overlap.

File: lib/maintainers.nix (hunk 1 above)
Comment (jtojnar) [reply lib/maintainers.nix]: `gavin` → `gavinrogers`

Comment (jtojnar) [reply lib/maintainers.nix]: `genesis` → `bignaux`

Comment (jtojnar) [reply lib/maintainers.nix]: `eelco` → `edolstra

**Comment by Profpatsch** [comment]
> That's why I suggested merging #34842 first

The problem with that commit is that lots of github handles are simply wrong, because the maintainer names don’t translate. So we are going to mention lots of non-nixos people.

File: lib/maintainers.nix (hunk 1 above)
Comment (bignaux) [reply lib/maintainers.nix]: ok

**Comment by jtojnar** [comment]
How hard would it be to check if given handle contributed to the nixpkgs repo and (manually) fix those that did not?

**Comment by 7c6f434c** [comment]
Well, may vanity counter does build a correspondence of names, emails and github names

File: lib/maintainers.nix (hunk 1 above)
Comment (7c6f434c) [inline lib/maintainers.nix]: That's actually me, 7c6f434c

Comment (jtojnar) [reply lib/maintainers.nix]: `raskin` → `7c6f434c`

Comment (jtojnar) [reply lib/maintainers.nix]: GitHub is really weird today.

**Comment by Profpatsch** [comment]
> How hard would it be to check if given handle contributed to the nixpkgs repo and (manually) fix those that did not?

Probably not very, might be sensible to use the Github API for that and rewrite the script (in the commit) a bit.

**Comment by jtojnar** [comment]
`curl "https://github.com/NixOS/nixpkgs/commits?author=$HANDLE" | grep 'No commits found'` should be good enough.

**Comment by Profpatsch** [comment]
Checking for commits right now, this finds quite a few accounts that are not commiters. `curl`ing the github repo is pretty slow, since github has very hard limits on the amount of requests possible (and returns HTTP `429` in that case).

**Comment by Profpatsch** [comment]
Okay, here we go. Now every handle that had no nixpkgs commits is corrected!

**Comment by Profpatsch** [comment]
Thanks to @aszlig’s input, I added a `shortName` property, which is used by hydra to send mails. @FRidh this should supersede #34842, minus the need for parsing.

**Comment by Profpatsch** [comment]
Removed the `shortName` again, @grahamc said we should fix hydra instead. According to @aszlig’s and my research, this will only lead to hydra not sending any mails for the time being.

**Comment by oxij** [comment]
Yay, nice!

Can we also rename this file to `maintainers/maintainers-list.nix` while we at this? Motivation: much saner `git log ./lib`. I wanted to do it for a while now, but was too lazy to make a separate PR for that.

File: lib/maintainers.nix [hunk 2]
```diff
@@ -1319,7 +1313,6 @@
   };
   fuzzy-id = {
     email = "hacking+nixos@babibo.de";
-    github = "fuzzy-id";
```
Comment (jtojnar) [inline lib/maintainers.nix]: https://github.com/NixOS/nixpkgs/pull/29147

File: lib/maintainers.nix [hunk 3]
```diff
@@ -1150,7 +1145,6 @@
   };
   epitrochoid = {
     email = "mpcervin@uncg.edu";
-    github = "epitrochoid";
```
Comment (jtojnar) [inline lib/maintainers.nix]: https://github.com/NixOS/nixpkgs/pull/9243

File: lib/maintainers.nix [hunk 4]
```diff
@@ -1020,7 +1017,6 @@
   };
   drewkett = {
     email = "burkett.andrew@gmail.com";
-    github = "drewkett";
```
Comment (jtojnar) [inline lib/maintainers.nix]: https://github.com/NixOS/nixpkgs/pull/14878

**Review by jtojnar — CHANGES_REQUESTED** [review]
Maybe also run `curl "https://github.com/NixOS/nixpkgs/pulls?utf8=%E2%9C%93&q=author%3A${user}" | grep 'No results matched your search'` for users that commit without matching e-mail address.

**Comment by Profpatsch** [comment]
@jtojnar I don’t understand, why is this a problem?

**Comment by Profpatsch** [comment]
Okay, since I brought it manually up-to-date with master, I’m going to merge now.
@jtojnar let’s put further discussion/changes into issues/more pull requests.

Don’t hesitate to revert/amend should this change break something important.

**Comment by oxij** [comment]
See #36275.

**Comment by Profpatsch** [comment]
@GrahamcOfBorg eval

**Comment by jtojnar** [comment]
@Profpatsch the problem is, you are removing GitHub handles that have commits in nixpkgs (but GitHub cannot pair them).

**Comment by Profpatsch** [comment]
I removed every handle where Github couldn’t resolve to a user page, yes, since those are either commits from non-Github users (via normal git commits) or Github accounts that don’t exist anymore.

**Comment by jtojnar** [comment]
At least the three I mentioned do resolve to a user page, are commits by GitHub users, and the accounts do still exist, only GitHub cannot pair them correctly (due to e-mail address mismatch). Checking the pull requests should fix these unpaired accounts.

Then only renamed accounts will remain. They need to be checked manually by checking the git log for expressions by the maintainer.

**Comment by Profpatsch** [comment]
Ah, I see. Can you open a new issue (or maybe even pull request) for those cases? Thanks for the scrutiny!

**Pull request merged**